plugins/outputs/influxdb_v2/http.go

```diff
@@ -201,12 +238,19 @@ func (c *httpClient) Write(ctx context.Context, metrics []telegraf.Metric) error
 		}
 
 		for bucket, batch := range batches {
+			if !c.SkipBucketCreation && !c.createBucketExecuted[bucket] {
+				if err := c.CreateBucket(ctx, bucket); err != nil {
+					log.Printf("W! [outputs.influxdb_v2] When writing to [%s]: bucket %q creation failed: %v\n", c.URL(), bucket, err)
```

Comment:
Doesn't it need to fail early in this case so it doesn't try to write the batch later to a nonexistent bucket? Or are there times when the code tries to create a bucket that already exists?

Do we need to approach bucket creation differently? Maybe we should try to write the batch first, watching for "bucket not found" errors. If there's no bucket then create the bucket and retry.

Reply:
Honestly, I'm not super sure why it's there. It's from the InfluxDB v1 plugin, which I was using as a reference.

My best guess is that it was to prevent unnecessary calls to create a database. However, I think it may have been accidentally circumvented since an attempt to create a database is made if a `DatabaseNotFoundError` was encountered. This would happen if the first `CreateDatabase` call failed with a 403 error since the database wouldn't have been created, but the `createDatabaseExecuted` entry would be `true`.

Moving all the `createBucketExecuted` stuff into `CreateBucket` and having it only create the bucket on when a `BucketNotFound` error is encountered would be a viable option. So then, as you suggested, we would always try to write the batch first, creating the bucket on any failure, and then retrying on the next flush interval..`